sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusConnection.cs

```diff
@@ -297,6 +298,17 @@ internal class ServiceBusConnection : IAsyncDisposable
             return builder.Uri.AbsoluteUri.ToLowerInvariant();
         }
 
+        /// <summary>
+        /// Throw an ObjectDisposedException if the object is Closing.
+        /// </summary>
+        internal virtual void ThrowIfClosed()
```

Comment:
You can use Argument.AssertNotDisposed for this.

Reply:
I just moved the order of the definition, to keep the `internal` members together.   I did not add this one.  :smile:

Reply:
Hehe, well the comment is still technically valid, but feel free to ignore 😄

Reply:
I'll take a peek when I loop back around and if it's not a bunch of churn, I'll clean it up.

Reply:
I changed the implementation of `ThrowIfClosed` over to delegate to `Argument.AssertNotDisposed` but did not remove it.   There was enough friction and a wide enough surface for the change that I didn't want to include it here.

While there are only a small number of call sites, there's some underlying architectural changes that are needed to remove things.  The tests mock the connection in multiple places and setup `ThrowIfClosed`, failing without it that setup to suppress the assertion failure.  There's no way to override the `IsClosed` member of the connection nor does the mock have a way to inject the inner type that `IsClosed` delegates to.

We probably don't want to mark `IsClosed` as `virtual`, so we'd need to either allow an explicit override for the member that supersedes delegating to the inner object or allow the inner object to be injected and it's `IsClosed` method mocked out.